src/TBuffer.cpp

```diff
@@ -1118,46 +1120,49 @@ void TBuffer::translateToPlainText(std::string& incoming, const bool isFromServe
-                                [[clang::fallthrough]];
+				mMXP_DEFAULT = mMXP_MODE = MXP_MODE_LOCKED;
+				break;
                             case 2: // locked line (until next newline) no MXP or HTML commands are allowed in the line.  The line is not parsed for any tags at all.  This is useful for "verbatim" text output from the MUD.  When a newline is received from the MUD, the mode reverts back to the Default mode.
-                                mMXP = false;
+				mMXP_MODE = MXP_MODE_LOCKED;
                                 break;
 
                             case 6: // lock secure mode (MXP 0.4 or later) - set secure mode.  Mode remains in effect until changed.  Secure mode becomes the new default mode.
-                                [[clang::fallthrough]];
+				mMXP_DEFAULT = mMXP_MODE = MXP_MODE_SECURE;
+				break;
                             case 1: // secure line (until next newline) all tags and commands in MXP are allowed within the line.  When a newline is received from the MUD, the mode reverts back to the Default mode.
-                                [[clang::fallthrough]];
+				mMXP_MODE = MXP_MODE_SECURE;
+				break;
                             case 4: // temp secure mode (MXP 0.4 or later) - set secure mode for the next tag only.  Must be immediately followed by a < character to start a tag.  Remember to set secure mode when closing the tag also.
-                                mMXP = true;
+				mMXP_MODE = MXP_MODE_TEMP_SECURE;
                                 break;
-
                             case 3: //  reset (MXP 0.4 or later) - close all open tags.  Set mode to Open.  Set text color and properties to default.
                                 closeT = 0;
                                 openT = 0;
-                                mAssemblingToken = false;
+				mAssemblingToken = false;
+				mMXP_MODE = mMXP_DEFAULT;
                                 currentToken.clear();
                                 mParsingVar = false;
                                 break;
-
                             case 5: // lock open mode (MXP 0.4 or later) - set open mode.  Mode remains in effect until changed.  OPEN mode becomes the new default mode.
-                                [[clang::fallthrough]];
+				mMXP_DEFAULT = mMXP_MODE = MXP_MODE_OPEN;
+				break;
                             case 0: // open line - only MXP commands in the "open" category are allowed.  When a newline is received from the MUD, the mode reverts back to the Default mode.  OPEN MODE starts as the Default mode until changes with one of the "lock mode" tags listed below.
-                                [[clang::fallthrough]];
+				mMXP_MODE = MXP_MODE_OPEN;
+				break;
                             default:
                                  if (modeCode <= 0 | modeCode == 5 | modeCode> 7) {
```

Comment:
You will probably want to update the tests and the comment here to reflect the changes you are making...